### PR TITLE
[[FIX]] Don't warn that an exported function is used before it is defined

### LINKED ISSUE #2658: latedef:true erroneously flags named exports as used before define
Version: 2.9.0, regressed from ~2.8.0

```
/*jshint latedef: true, esnext: true*/
export function hi() {
 return 1;
}
```

results in:

```
2	'hi' was used before it was defined.
```

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
Fixes #2658

### CODE REVIEW

**Comment by lukeapage** [comment]
LGTM

File: tests/unit/fixtures/latedef-esnext.js [hunk 1]
```diff
@@ -60,3 +60,8 @@ let dc = {
         return dc;
     }
 };
+
+export function de() {}
+export class df {}
+export var dg;
+export let dh;
```
Comment (mikesherov) [inline tests/unit/fixtures/latedef-esnext.js]: Maybe add one for const?

Comment (lukeapage) [inline tests/unit/fixtures/latedef-esnext.js]: yes, even though let and const now share the same code, there is no reason to leave it off, sorry I missed that.

Comment (nicolo-ribaudo) [inline tests/unit/fixtures/latedef-esnext.js]: Done

**Comment by mikesherov** [comment]
LGTM

File: tests/unit/fixtures/latedef-esnext.js [hunk 2]
```diff
@@ -60,3 +60,9 @@ let dc = {
         return dc;
     }
 };
+
```
Comment (jugglinmike) [inline tests/unit/fixtures/latedef-esnext.js]: Could you add a comment here referencing the regression at gh-2658 so future contributors understand the purpose of these statements?

**Comment by jugglinmike** [comment]
@nicolo-ribaudo This looks good! Just one documentation-related thing, then this should be good to go.

File: tests/unit/fixtures/latedef-esnext.js (hunk 2 above)
Comment (nicolo-ribaudo) [inline tests/unit/fixtures/latedef-esnext.js]: Done.

**Comment by jugglinmike** [comment]
Squashed and merged to `master` at commit d0433d23e1a48c74522b9ed32b87c45624bef4c5. Thanks, @nicolo-ribaudo!

**Pull request closed without merging**